### PR TITLE
feat: add alpha parameter to `lasso_regression`

### LINKED ISSUE #163: Set alpha parameter for regularization of `LassoRegression`
### Is your feature request related to a problem?

It's not possible to set the alpha parameter for regularization of lasso regression.

### Desired solution

* Add a new parameter `alpha: float` to the initializer of `safeds.ml.regression.LassoRegression`
* Raise a value error if `alpha` < 0
* Warn if `alpha == 0` -> Users should use `LinearRegression` instead
* Pass it along to `alpha` of the wrapped `scikit-learn` model in the `fit` method

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

* Use `warnings.warn` to warn.
* Use `with pytest.warns` to test that the warning is created.

### PR DESCRIPTION
**Pull request by alex-senger**
Closes #163.

### Summary of Changes

Add alpha parameter to lasso regression

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.1) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/4764954869)

| Descriptor  |                                 Linter                                 |Files|Fixed|                            Errors                            |Elapsed time|
|-------------|------------------------------------------------------------------------|-----|----:|--------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/6.22.1/descriptors/python_black)          |    2|    0|                                                             0|0.56s       |
|✅ PYTHON    |[mypy](https://megalinter.io/6.22.1/descriptors/python_mypy)            |    2|     |                                                             0|1.57s       |
|❌ PYTHON    |[ruff](https://megalinter.io/6.22.1/descriptors/python_ruff)            |    2|    2|[4](https://github.com/Safe-DS/Stdlib/actions/runs/4764954869)|0.04s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.1/descriptors/repository_git_diff)|yes  |     |no                                                            |0.02s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/4764954869)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

**Comment by alex-senger** [comment]
Note that pytest isn't catching the warning correctly.

File: src/safeds/ml/classical/regression/_lasso_regression.py [hunk 1]
```diff
@@ -13,9 +14,25 @@
...
+    ------
+    ValueError
+        If alpha is negative.
+    """
+
+    def __init__(self, alpha: float = 1.0) -> None:
+        if alpha < 0:
+            raise ValueError("alpha must be non-negative")
+        if alpha == 0:
+            warn("alpha is zero, you should use LinearRegression instead", UserWarning, stacklevel=1)
```
Comment (lars-reimann) [inline src/safeds/ml/classical/regression/_lasso_regression.py:34]:
```suggestion
            warn("Setting alpha to zero makes this model equivalent to LinearRegression. You should use LinearRegression instead for better numerical stability.", UserWarning, stacklevel=1)
```

I think this better explains why this is an issue.

File: src/safeds/ml/classical/regression/_lasso_regression.py [hunk 2]
```diff
@@ -13,9 +14,25 @@
...
 class LassoRegression(Regressor):
-    """Lasso regression."""
-
-    def __init__(self) -> None:
+    """Lasso regression.
+
+    Parameters
+    ----------
+    alpha : float, default=1.0
+        Constant that multiplies the L1 term. Defaults to 1.0.
```
Comment (lars-reimann) [inline src/safeds/ml/classical/regression/_lasso_regression.py:22]:
```suggestion
    alpha : float
        Controls the regularization of the model. The higher the value, the more regularized it becomes.
```

This does not rely on the mathematical definition of lasso regression as much. If we add the formular in the future, we can add your explanation again.

We usually don't need to repeat the default value here.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

File: tests/safeds/ml/classical/regression/test_lasso_regression.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import pytest
+from safeds.data.tabular.containers import Table
+from safeds.ml.classical.regression import LassoRegression
+
+
+def test_alpha_invalid() -> None:
```
Comment (lars-reimann) [inline tests/safeds/ml/classical/regression/test_lasso_regression.py:6]: We should also use the `test_should_` convention for these tests.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

**Review by lars-reimann — APPROVED** [review]
LGTM, thanks!

**Pull request merged**